ql/src/java/org/apache/hadoop/hive/ql/parse/MetaDataExportListener.java

```diff
@@ -90,11 +92,9 @@ private void export_meta_data(PreDropTableEvent tableEvent) throws MetaException
       EximUtil.createExportDump(fs, outFile, mTbl, null, null,
           new HiveConf(conf, MetaDataExportListener.class));
       if (moveMetadataToTrash == true) {
-        wh.deleteDir(metaPath, true);
+        wh.deleteDir(metaPath, true, Hive.get().getDatabase(tbl.getDbName()));
       }
-    } catch (IOException e) {
-      throw new MetaException(e.getMessage());
-    } catch (SemanticException e) {
+    } catch (IOException | MetaException | HiveException e) {
```

Comment:
Wy wrapping MetaException over another MetaException. Shall remove it from catch block.